### DISCUSSION TITLE
How to "convert" a request into a 404?

### DISCUSSION THREAD

**Discussion by PieterBoeren** [dotnet/yarp#593] [Q&A] [answered]
I'm proxying multiple micro-frontends and api's in the proxy, but I have some issues with it. In order to get the micro-frontends working, I defined a catch all and let that redirect to some cluster. However, if I now navigate to something like "/api/notdefinedinproxy", I get a HTML response instead of some kind of error (all logical of course, the catch all causes this).

I tried adding a special proxy configuration like "api/{*anything}" which works more or less, but I still have to define a cluster with a destination (if I remember correctly, you'll get some kind of 50x error).
I tried solving that with some custom code, but I couldn't figure out how to achieve this:

```
endpoints.MapReverseProxy(applicationBuilder =>
{
    applicationBuilder.Use(async (context, next) =>
    {
        var routeName = context.GetEndpoint().DisplayName;

        if (routeName.Equals("routeApiFallback", StringComparison.OrdinalIgnoreCase))
        {
            context.Response.StatusCode = StatusCodes.Status404NotFound;
            context.Abort();
        }
        else
        {
           // Here I'm doing some other stuff
        }

        await next().ConfigureAwait(false);
    });
});
```

Is this the only way my requirement is going to work, or is there another way? And what is the best way of returning that 404 (the above code is not working entirely)?

**Comment by Tratcher** [comment, marked as answer]
How about using a standard route instead of a proxy route?
```
endpoints.Map("/api/{*anything}", context => { context.Response.StatusCode = StatusCodes.Status404NotFound; return Task.CompletedTask; });
```

By the way, middleware should only call Abort if they don't want the response to be sent. If they do want it to be sent then they should return early / not call next.

**Reply by PieterBoeren** [reply]
That makes sense, thanks.
